cmd/artifact/install/install.go

```diff
@@ -232,10 +255,21 @@ func (o *artifactInstallOptions) RunArtifactInstall(ctx context.Context, args []
 	o.Printer.Info.Printfln("Installing the following artifacts: %v", refs)
 
 	for _, ref := range refs {
+		var sig *index.Signature
 		ref, err = o.IndexCache.ResolveReference(ref)
 		if err != nil {
 			return err
 		}
+		sig, ok := signatures[ref]
+		if !ok {
+			// try to get the signature from the index
+			o.IndexCache.SignatureForIndexRef(ref)
+		}
+
+		repo, err := utils.RepositoryFromRef(ref)
+		if err != nil {
+			return err
+		}
```

Comment:
Could we move this logic inside the `if block` that runs when we need to verify the artifact?

Reply:
The repo part: yes
The signature part: I prefer having it before the `if` block because this way, if we have a signature to verify we're going to output the `"Verifying signature for %s"` log line, otherwise we won't.